lib/services/schedule.js

```diff
@@ -0,0 +1,222 @@
+    throw new Error('Client: Missing "at" property as number.');
+  } else if (!references.isUrl(publish)) {
+    throw new Error('Client: Missing "publish" property as valid url.');
+  }
+
+  return `${prefix}/_schedule/${buf.encode(publish.replace(/https?:\/\//, ''))}`;
+}
+
+/**
+ * NOTE:  We _cannot_ delete without knowing the thing that was published because we need to delete
+ * the @scheduled location as well.
+ *
+ * @param {string} uri
+ * @param {object} user
+ * @returns {Promise}
+ */
+function del(uri, user = {}) {
+  return db.selectItem(uri)
+    .then((response) => {
+      const data = response.rows.length ? response.rows[0].data : {},
+        publish = data[publishProperty];
+      let pageUri;
+
+      if (publish) {
+        pageUri = publish.replace(/http:\/\/|https:\/\//g, '');
+
+        return db.deleteItem(uri)
+          .then(() => db.db.getMeta(pageUri))
+          .then((meta) => {
+            // update the page's metadata
+            meta.scheduled = false;
+            meta.scheduledTime = null;
+            meta.history.push({
+              action: 'unschedule',
+              users: [user],
+              timestamp: moment().toISOString()
+            });
+
+            return db.db.patchMeta(pageUri, meta);
+          })
+          .then(() => data);
+      } else {
+        return Promise.resolve(data); // TODO: Some meaningful message
+      }
+    }).catch(errorLogger);
+}
+
+/**
+ * Create a schedule item to publish something in the future
+ * @param {string} uri
+ * @param {object} data
+ * @param {object} user
+ * @returns {Promise}
+ */
+function post(uri, data, user) {
+  const reference = createScheduleObjectKey(uri, data),
+    at = data[scheduledAtProperty],
+    publish = data[publishProperty],
+    pageUri = publish.replace(/http:\/\/|https:\/\//g, '');
+
+  // first, insert data into schedule table
+  return db.insertItem(reference, data)
+    .then(() => db.db.getMeta(pageUri)) // then, update the page's metadata
+    .then((meta) => {
+      meta.scheduled = true;
+      meta.scheduledTime = moment(at).toISOString();
+      meta.history.push({
+        action: 'schedule',
+        users: [user],
+        timestamp: moment().toISOString()
+      });
+
+      return db.db.patchMeta(pageUri, meta);
+    })
+    .then(() => data)
+    .catch(errorLogger);
+}
+
+/**
+ * Gets a scheduled item.
+ * @param {string} uri
+ * @return {Promise}
+ */
+function getScheduleItem(uri) {
+  return db.selectItem(uri)
+    .then((data) => data.rows.length ? data.rows[0] : {})
+    .catch(errorLogger);
+}
+
+/**
+ * Gets the scheduled items of a site.
+ * @param {string} uri
+ * @return {Promise<T | void>}
+ */
+function getScheduleList(uri) {
+  const prefix = getPrefix(uri);
+
+  return db.selectItemsFromSite(prefix)
+    .then((data) => data.rows)
+    .catch(errorLogger);
+}
+
+/**
+ * Starts waiting for things to publish (but only if we're not already listening)
+ */
+function startListening() {
```

Comment:
I would like to move the env var to initialize the listener to wrap this function.

Reply:
already covered [here](https://github.com/clay/amphora-schedule/pull/1/files#diff-d2a1ca47cb16a2758bdea924a371c5d5R17)